text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+  about portability to their target, and enhance their documentation of the
+  target.
+- The target must build reliably in CI.
+- Building the target must not take substantially longer than other targets.
+- Tier 2 targets must support building on the existing targets used for CI
+  infrastructure.
+- Tier 2 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to ensure that tests pass for the target.
+  In particular, do not post comments (automated or manual) on a PR that
+  suggests a block on the PR based on tests failing for the target. (A PR
+  author must not break the build of a tier 2 target, but need not ensure the
+  tests pass for a tier 2 target, even if notified that they fail.)
+- The target development team should regularly run the testsuite for the
+  target, and should fix any test failures in a reasonably timely fashion.
+- A tier 2 target may be demoted or removed if it no longer meets these
+  requirements. Any proposal for demotion or removal will be CCed to people who
+  have previously worked on the platform, and will be communicated widely to
+  the Rust community before being dropped from a stable release.
+- All tier 3 requirements apply.
+
+Note: some tier 2 platforms additionally have binaries built to run on them as
+a host (such as `rustc` and `cargo`). Such a platform must meet all the
+requirements above, and must additionally get the compiler and infrastructure
+team to approve the building of host tools.
+
+## Tier 1 target policy
+
+At this tier, the Rust project guarantees that a target builds and passes all
+tests, and will reject patches that fail to build or pass the testsuite on a
+target. We hold tier 1 targets to our highest standard of requirements.
+
+- Any new tier 1 target requires compiler team approval based on these
+  requirements.
+- In addition, the infrastructure team must approve the integration of the
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- In addition, the release team must approve the long-term viability of the
+  target, and the additional work of supporting the target.
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. A tier 1 target may
+  be demoted or removed if it becomes obsolete or no longer meets this
+  requirement.
+- The target must build and pass tests reliably in CI.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as systems
+  running the target), the target development team shall arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
```

Comment:
This seems surprising and possibly contradictory. I would expect that targets would need to be tier 2 before they can become tier 1, but the tier 2 list specifically says that such targets must not require new infrastructure, so I'm not sure how a target could ever be considered for tier 1 status without its build infrastructure already existing. Perhaps this bullet should be moved to the tier 2 list, and modified to indicate that the requirement for new infrastructure *may* be taken as a reason to reject a new tier 2 target?

(I suppose one possible case is that the target can be _built_ without new infrastructure but not adequately _tested_, in which case it would be possible to make a target tier-2 but not tier-1.)

Reply:
For Tier 2 we can just cross-compile from Linux/Windows/macOS to most of the targets, while running the test suite requires to have that target as the host platform. So I expect this scenario to happen for most targets.